#! /bin/sh /usr/share/dpatch/dpatch-run
## 42_alpha_fetch_and_add.dpatch by Bradley Hughes <bhughes@trolltech.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fetch-and-add is supposed to return the original value before the addition.

@DPATCH@
diff -urNad qt4-x11-4.3.0~/src/corelib/arch/qatomic_alpha.h qt4-x11-4.3.0/src/corelib/arch/qatomic_alpha.h
--- qt4-x11-4.3.0~/src/corelib/arch/qatomic_alpha.h	2007-05-25 15:24:09.000000000 +0200
+++ qt4-x11-4.3.0/src/corelib/arch/qatomic_alpha.h	2007-07-16 21:52:20.000000000 +0200
@@ -190,7 +190,7 @@
                  : "=&r" (old), "=&r" (tmp), "+m"(*ptr)
                  : "r" (value)
                  : "memory");
-    return old != -1;
+    return old;
 }
 
 inline int q_atomic_fetch_and_add_acquire_int(volatile int *ptr, int value)
@@ -208,7 +208,7 @@
                  : "=&r" (old), "=&r" (tmp), "+m"(*ptr)
                  : "r" (value)
                  : "memory");
-    return old != -1;
+    return old;
 }
 
 inline int q_atomic_fetch_and_add_release_int(volatile int *ptr, int value)
@@ -226,7 +226,7 @@
                  : "=&r" (old), "=&r" (tmp), "+m"(*ptr)
                  : "r" (value)
                  : "memory");
-    return old != -1;
+    return old;
 }
 
 #else // !Q_CC_GNU
